### PR TITLE
docs: Add onboarding docs for cluster-scoped resources

### PR DESCRIPTION
**Pull request by tumido**
Requires: https://github.com/operate-first/apps/pull/120, 
Related: https://github.com/operate-first/odh-moc-support/issues/39, https://github.com/operate-first/blueprint/pull/19

A single issue template for Onboarding to the cluster management + a guide on how to do it yourself. A issue template for ArgoCD will follow.

### CODE REVIEW

**Review by anishasthana — APPROVED** [review]
This looks good to me.

**Comment by tumido** [comment]
@anishasthana can you please take one more look? I've updated it with the secret management and how to add a team specific GPG key + made the issue template more fool proof with target environment specification and handing over of the GPG key. :wink:

File: .github/ISSUE_TEMPLATE/onboarding_to_cluster.md
Comment (anishasthana) [inline .github/ISSUE_TEMPLATE/onboarding_to_cluster.md:6]: Shouldn't assignees actually be empty? This way any two OWNERS of the repo can onboard users instead of being blocked on just the two of you.

File: .github/ISSUE_TEMPLATE/onboarding_to_cluster.md
Comment (anishasthana) [inline .github/ISSUE_TEMPLATE/onboarding_to_cluster.md:23]: Can we add samples of what the namespaces names could look like?

File: docs/onboarding_to_cluster.md
Comment (anishasthana) [inline docs/onboarding_to_cluster.md:13]: Question: We also have a toolbox available on github right? Should we instead tell people to use that?

File: docs/onboarding_to_cluster.md
Comment (anishasthana) [inline docs/onboarding_to_cluster.md:48]: /s/cluser/cluster

File: docs/onboarding_to_cluster.md
Comment (anishasthana) [inline docs/onboarding_to_cluster.md:65]: /s/bellow/below

**Review by anishasthana — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/onboarding_to_cluster.md
Comment (tumido) [reply .github/ISSUE_TEMPLATE/onboarding_to_cluster.md:6]: Will change that! :+1: Good point

File: docs/onboarding_to_cluster.md
Comment (tumido) [reply docs/onboarding_to_cluster.md:13]: Good point! We should offer it as the default option!

**Review by anishasthana — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/onboarding_to_cluster.md
Comment (HumairAK) [inline .github/ISSUE_TEMPLATE/onboarding_to_cluster.md:11]: change to: 

> Select all that apply:

File: docs/onboarding_to_cluster.md
Comment (HumairAK) [inline docs/onboarding_to_cluster.md:1]: Can we add a brief summary of what this doc accomplishes, should be a short one or two liner.

File: docs/onboarding_to_cluster.md
Comment (HumairAK) [inline docs/onboarding_to_cluster.md:11]: The numbering here seems to imply that they should do 1, then 2, but it's actually 1 or 2, to give this impression I would format it like so: 

> You will need pre-requisite tools to follow along with this doc, please do either of the following: 
> - Install our toolbox to have the developer setup ready automatically for you.
> - Install the tools manually. You'll need kustomize, sops and ksops.

File: docs/onboarding_to_cluster.md
Comment (HumairAK) [inline docs/onboarding_to_cluster.md:44]: Can we add a one liner that says something like the following: 

"The following steps enable users to access designated cluster/namespaces. This simply consists of adding such users to the appropriate OpenShift groups with access to those cluster/namespaces. These groups are encrypted to protect user sensitive data."

File: docs/onboarding_to_cluster.md
Comment (HumairAK) [inline docs/onboarding_to_cluster.md:8]: a conditional pre-requisite  is a gpg key & fingerprint of that key if they are adding a new group for that cluster.

File: docs/onboarding_to_cluster.md
Comment (HumairAK) [inline docs/onboarding_to_cluster.md:54]: What do you mean by team denotation?

File: docs/onboarding_to_cluster.md
Comment (HumairAK) [reply docs/onboarding_to_cluster.md:8]: although I'm noticing that these are included in the issue-template, I'll leave it to your discretion whether we should mention that here. Also worth considering are things like 

- Cluster -> namespace(s) to add
- user(s) to add

File: docs/onboarding_to_cluster.md
Comment (HumairAK) [inline docs/onboarding_to_cluster.md:59]: I think a brief one line comment explaining this entry would be useful, something like: 

> Adding the following entry will encrypt a team's group on the designated cluster. The GPG is provided by the team allowing them to decrypt and make changes as needed.

File: docs/onboarding_to_cluster.md
Comment (HumairAK) [inline docs/onboarding_to_cluster.md:60]: I'm a little confused by the path_regex here, we haven't been using prefixes, should we be using them? As it stands, the group manifests are  are just named after a designated team name. E.g. for operate-first we say `operate-first.enc.yaml` and not `opf-group.enc.yaml`. We could just say "cluster-scope/.*YOUR_TEAM_NAME.*" and further explain that `YOUR_TEAM_NAME` must match the group manifest name.

File: docs/onboarding_to_cluster.md
Comment (HumairAK) [inline docs/onboarding_to_cluster.md:116]: Let's add another line here like: 

"Then simply append the list of users with the new users that need to be added."

File: docs/onboarding_to_cluster.md
Comment (HumairAK) [inline docs/onboarding_to_cluster.md:22]: Can we add a comment about, when picking a namespace, use a prefix for all their namespaces that's unique to their team. This will make it easier for argocd multi tenancy later.

File: docs/onboarding_to_cluster.md
Comment (tumido) [reply docs/onboarding_to_cluster.md:8]: Yeah, I've listed only the technical requirements. I'll add that. LMKWYT

File: docs/onboarding_to_cluster.md
Comment (tumido) [reply docs/onboarding_to_cluster.md:44]: Copy pasted with a slight changes to the PR. :smile:

File: docs/onboarding_to_cluster.md
Comment (tumido) [reply docs/onboarding_to_cluster.md:60]: Yeah.. I wanted to have a single rule per team (which would be solved by the prefix), but I think it may not be the best idea to have these sops rules anymore at all... They are just too confusing and also their order matters (the first resolved rule gets used)...

I think this is relevant to the comment you made above as well, I've added the rule explanation, but I'm not satisfied with it very much.

What about we scratch the idea of adding rules to `.sops.yaml` all together and just encrypt the files with CLI?

```
sops -e --encrypted-regex="^users$" --pgp="0508677DD04952D06A943D5B4DC4116D360E3276, YOUR_TEAM_GPG_KEY" path/to/resource.yaml
```

The encrypted resource hold the encryption metadata anyways so any further edit on that file will keep the `encrypted_regex` the same as well as the used keys... WDYT?

File: docs/onboarding_to_cluster.md
Comment (tumido) [reply docs/onboarding_to_cluster.md:54]: I meant the prefix and I don't like it anymore. See bellow, please.

File: docs/onboarding_to_cluster.md
Comment (goern) [inline docs/onboarding_to_cluster.md:116]: these `'` look like a type?!

Comment (tumido) [reply docs/onboarding_to_cluster.md]: changed to the less typo looking `"`

File: docs/onboarding_to_cluster.md
Comment (HumairAK) [reply docs/onboarding_to_cluster.md:82]: Yeah, this makes it easier to digest. Let's go with this.

Comment (tumido) [reply docs/onboarding_to_cluster.md:82]: Discussed offline. Changing the docs to the solution described in the [last comment](https://github.com/operate-first/support/pull/40#discussion_r560835605)

**Review by HumairAK — APPROVED** [review]

**Pull request merged**
